### PR TITLE
Recommend panic::resume_unwind instead of panicking.

### LINKED ISSUE #79950: Update `std::thread::Result` and `std::panic::catch_unwind` documentation to recommend `std::panic::resume_unwind` instead of `unwrap`
The current version of the `std::thread::Result` documentation recommends unwrapping a `Result<T, Box<dyn Any + ...>>` in order to re-start propagation of a caught panic. This recommendation will not simply resume the panic as it was originally. Once the error type has been erased and Boxed the panic can no longer be downcast directly back to the actual payload type. Instead you have to also try downcasting twice.

`std::panic::catch_unwind` on the other hand has no references to `resume_unwind` or how to repropagate caught panics at all, which ends up pushing users into the `std::thread::Result` if they're trying to figure out how to handle the `Err` variant of `catch_unwind`.

These docs should reference `resume_unwind` as the recommended way for re-propagating type erased panic payloads.

### PR DESCRIPTION
**Pull request by frewsxcv**
Fixes https://github.com/rust-lang/rust/issues/79950.

### CODE REVIEW

File: library/std/src/thread/mod.rs [hunk 1]
```diff
@@ -1186,32 +1186,37 @@ impl fmt::Debug for Thread {
 /// the [`Error`](crate::error::Error) trait.
 ///
 /// Thus, a sensible way to handle a thread panic is to either:
-/// 1. `unwrap` the `Result<T>`, propagating the panic
+///
+/// 1. propagate the panic with [`std::panic::resume_unwind`]
 /// 2. or in case the thread is intended to be a subsystem boundary
 /// that is supposed to isolate system-level failures,
-/// match on the `Err` variant and handle the panic in an appropriate way.
+/// match on the `Err` variant and handle the panic in an appropriate way
```
Comment (gThorondorsen) [inline library/std/src/thread/mod.rs:1193]: Why remove the period? It's a complete sentence even though it is formatted with an enumeration.

File: library/std/src/thread/mod.rs [hunk 2]
```diff
@@ -1186,32 +1186,37 @@ impl fmt::Debug for Thread {
...
 ///
 /// A thread that completes without panicking is considered to exit successfully.
 ///
 /// # Examples
 ///
+/// Matching on the result of a joined thread:
+///
```
Comment (gThorondorsen) [inline library/std/src/thread/mod.rs:1200]: This seems to mean that the example is going to show case (2) above. However, the example below is changed from case (2) to case (1). Which one is correct/intended? Or should there be two examples?

File: library/std/src/thread/mod.rs (hunk 1 above)
Comment (frewsxcv) [reply library/std/src/thread/mod.rs:1193]: The statement on its own is not a complete sentence

File: library/std/src/thread/mod.rs (hunk 2 above)
Comment (frewsxcv) [reply library/std/src/thread/mod.rs:1200]: Previously it was just demonstrating matching. Now it is demonstrating matching and the panic_unwind.

**Pull request merged**